src/components/SuggestionsList/SuggestionsListItem.tsx

```diff
@@ -0,0 +1,68 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
+import MediaObject, { MediaObjectSize } from '../MediaObject';
+import Avatar, { AvatarSize } from '../Avatar';
+import './SuggestionsListItem.css';
+
+export interface SuggestionItem {
+  id: string | number;
+  imageUrl?: string;
+  name: string;
+  description?: string;
+}
+
+export interface SuggestionsListItemProps extends SuggestionItem, NestableBaseComponentProps {
+  searchText: string;
+  isHovered: boolean;
+  isSelected: boolean;
+  onHover(id: string | number): void;
+  onSelect(item: SuggestionItem): void;
+}
+
+const baseClass = 'y-suggestions-list-item';
+const selectedClass = `${baseClass} y-suggestions-list-item--selected`;
+const hoveredClass = `${baseClass} y-suggestions-list-item--hover`;
+const highlightedClass = `y-suggestions-list-item--highlight`;
+
+const getHighlightedName = (name: string, search: string) => {
+  return name.split(new RegExp(`(${search})`, 'gi')).map((item: string, index: number) => {
+    const className = search.toLowerCase() === item.toLowerCase() ? highlightedClass : undefined;
+    return <span key={index} className={className}>{item}</span>;
+  });
+};
+
+export default class SuggestionsListItem extends React.PureComponent<SuggestionsListItemProps, {}> {
+  public render() {
+    const { isSelected, isHovered, name, searchText, imageUrl, description } = this.props;
+    const avatar = (
+      <Avatar
```

Comment:
this assumes results are all users which display as round avatar in a MediaObject. groups need `borderType: SOFT`, other results with an imageUrl would use an `<Image>` component instead of Avatar for square borders. then there are plain text results which have no image, take up less vertical space, and align all the way to the left instead of using MediaObject.

in the spec meeting we were thinking SuggestionListItem could have a template option. the application would own the business logic of mapping a result type to one of the supported templates:

`type PickerItemTemplate = 'text' | 'mediaRound' | 'mediaSoft' | 'mediaSquare';`

Reply:
yup - i'm working along the critical path of my project. In the future when other types of results can be rendered this component can be updated